src/zelos/tools/zdbserver/zdbserver.py

```diff
@@ -0,0 +1,340 @@
+                "end_address": f"0x{region.end:x}",
+            }
+            vmmap.append(entry)
+        return vmmap
+
+    def run(self) -> dict:
+        """
+        Run until a break condition is encountered.
+
+        Returns:
+            Information about the break condition in the format:
+
+            .. code-block:: python
+
+                break_state = {
+                    'pc': INT,
+                    'syscall': {
+                        'name': STR,
+                        'args': [
+                            { 'type': STR, 'name': STR, 'value': INT },
+                            ...
+                        ],
+                        'retval': INT,
+                        'retval_register': STR,
+                    },
+                    'bits': INT,
+                }
+
+        """
+        logger.debug(f"[debug] run()")
+        break_state = self.z.start()
+        # Format break_state for RPC by converting `None` and Numbers to
+        # RPC-serializable values.
+        if break_state is None:
+            return {}
+        break_state["pc"] = "0x{0:x}".format(break_state["pc"])
+        if break_state.get("syscall", None) is None:
+            break_state["syscall"] = {}
+        else:
+            break_state["syscall"]["retval"] = "0x{0:x}".format(
+                break_state["syscall"]["retval"]
+            )
+            for arg in break_state["syscall"]["args"]:
+                arg["value"] = "0x{0:x}".format(arg["value"])
+        return break_state
+
+    def stop(self) -> bool:
+        """
+        Stop running.
+
+        Returns:
+            True if no exception occurred.
+        """
+        logger.debug(f"[debug] stop()")
+        self.z.stop("debug stop")
+        return True
+
+    def set_syscall_breakpoint(self, name: str) -> bool:
+        """
+        Set a breakpoint at all syscalls of a specified name. Breaks
+        will occur *after* the syscall has completed and PC has
+        advanced to the next instruction.
+
+        Args:
+            name: Target syscall to set breakpoint at.
+
+        Returns:
+            True if no exception occurred.
+        """
+        logger.debug(f"[debug] set_syscall_breakpoint({name})")
+        self.z.set_syscall_breakpoint(name)
+        return True
+
+    def remove_syscall_breakpoint(self, name: str) -> bool:
+        """
+        Remove a previously set syscall breakpoint specified by name.
+
+        Args:
+            name: Target syscall to remove breakpoints from.
+
+        Returns:
+            True if no exception occurred.
+        """
+        logger.debug(f"[debug] remove_syscall_breakpoint({name})")
+        self.z.remove_syscall_breakpoint(name)
+        return True
+
+    def get_filepath(self) -> str:
+        """
+        Get the file path of the emulated binary.
+
+        Returns:
+            The full path to the emulated binary.
+        """
+        return self.z.internal_engine.main_module_name
```

Comment:
After #58 is merged, you can replace with the updated API

`return self.z.main_module_path`

Reply:
done